server/src/linter.ts

```diff
@@ -0,0 +1,198 @@
+import { spawn } from 'child_process'
+import * as LSP from 'vscode-languageserver'
+
+function formatMessage(comment: ShellcheckComment): string {
+  return (comment.code ? `SC${comment.code}: ` : '') + comment.message
+}
+
+type LinterOptions = {
+  executablePath?: string | null
```

Comment:
Let us drop the undefined option as `getShellcheckPath` always returns `string | null`

```suggestion
  executablePath: string | null
```